doc/reference/kernel/other/event_manager.rst

```diff
@@ -0,0 +1,359 @@
+.. _event_manager:
+
+Event Manager
+#############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Event Manager is a piece of software that supports development of consistent, modular, event-based application.
+In an event-based application, parts of the application functionality are separated into isolated modules that communicate with each other using events.
+Events are submitted by modules and other modules can subscribe and react to them.
+The Event Manager acts as coordinator of the event-based communication.
+
+.. figure:: em_overview.svg
+   :alt: Architecture of an application based on Event Manager
+
+See the :ref:`event_manager_sample` sample for a simple example of how to use the Event Manager.
+
+Events
+******
+
+Events are structured data types that are defined by the application and can contain additional data.
+
+The Event Manager handles the events by processing and propagating all of them to the modules (listeners) that subscribe to a specific event.
+Multiple modules can subscribe to the same event.
+As part of this communication, listeners can process events differently based on their type.
+
+The Event Manager provides API for defining, creating, and subscribing events.
+See `Implementing an event type`_ for details about how to create custom event types.
+
+Modules
+*******
+
+Modules are separate source files that can subscribe to every defined event.
+You can use events for communication between modules.
+
+There is no limitation as to how many modules each module can subscribe to.
+An application can have as many modules as required.
+
+The Event Manager provides API for subscribing modules to a specific events defined in the application.
+When a module is subscribing to a specific event, it is called a listener module.
+Every listener is identified by a unique name.
+
+.. _event_manager_configuration:
+
+Configuration
+*************
+
+To use Event Manager, you must enable the following Kconfig options:
+
+* :kconfig:`CONFIG_EVENT_MANAGER` - This option enables the Event Manager.
+* :kconfig:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` - This option enables orphan memory sections used by the Event Manager.
+  Set this option to suppress warnings and errors.
+
+Initializing the Event Manager
+==============================
+
+You must initialize the Event Manager in your :file:`main.c` file.
+Initializing the Event Manger allows it to handle submitted events and deliver them to modules that subscribe to the specified event type.
+
+To initialize the Event Manager, complete the following steps:
+
+1. Include :file:`event_manager.h` in your :file:`main.c` file.
+#. Call :c:func:`event_manager_init()`.
```

Comment:
why does event manager need to be started manually? could it be done via SYS_INIT?

Reply:
The event manager was created to be an application skeleton (not a system one) and I believe it is better for application to have a control when event processing is started.
This said initialization function does almost nothing at the moment - it performs checks but was left in case we would like to extend the configuration in the future (like start a separate event processing thread, prepare an allocator or something similar).